src/java.base/share/classes/java/util/UUID.java

```diff
@@ -485,73 +485,79 @@ public String toString() {
-     *         (carry_mask << 1) + (carry_mask >> 1) - (carry_mask >> 4) = 0x27
-     *    c. Parallel Addition: Combines base/letter adjustment with original nibbles
-     *
-     * 3. Byte Order Correction:
-     *    - Reverses byte order to match big-endian string representation
-     *
-     * Performance Considerations:
-     * - Processes all 8 nibbles simultaneously using bitwise operations (~1 cycle)
-     * - Eliminates:
-     *   - Loop overhead (traditional 8-iteration approach)
-     *   - Branch mispredictions (no if/else for letter/digit)
-     *   - Multiple memory accesses (works entirely in registers)
-     * - Achieves O(1) time complexity vs O(n) for iterative approaches
-     * - Uses 6 arithmetic/logical operations vs 40+ for naive implementation
-     *
-     * @param i The input value containing up to 4 bytes of data in its least significant bits
-     * @return A long value representing the concatenated hexadecimal string of the 4 input bytes
+     * Efficiently converts 8 hexadecimal digits to their ASCII representation using SIMD-style vector operations.
+     * This method processes multiple digits in parallel by treating a long value as eight 8-bit lanes,
+     * achieving significantly better performance compared to traditional loop-based conversion.
+     *
+     * <p>The conversion algorithm works as follows:
+     * <pre>
+     * 1. Input expansion: Each 4-bit hex digit is expanded to 8 bits
+     * 2. Vector processing:
+     *    - Add 6 to each digit: triggers carry flag for a-f digits
+     *    - Mask with 0x10 pattern to isolate carry flags
+     *    - Calculate ASCII adjustment: (carry << 1) + (carry >> 1) - (carry >> 4)
+     *    - Add ASCII '0' base (0x30) and original value
+     * 3. Byte order adjustment for final output
+     * </pre>
+     *
+     * <p>Performance characteristics:
+     * <ul>
+     *   <li>Processes 8 digits in parallel using vector operations
+     *   <li>Avoids branching and loops completely
+     *   <li>Uses only integer arithmetic and bit operations
+     *   <li>Constant time execution regardless of input values
+     * </ul>
+     *
+     * <p>ASCII conversion mapping:
+     * <ul>
+     *   <li>Digits 0-9 → ASCII '0'-'9' (0x30-0x39)
+     *   <li>Digits a-f → ASCII 'a'-'f' (0x61-0x66)
+     * </ul>
+     *
+     * @param input A long containing 8 hex digits (each digit must be 0-15)
+     * @return A long containing 8 ASCII bytes representing the hex digits
+     *
+     * @implNote The implementation leverages CPU vector processing capabilities through
+     *           long integer operations. The algorithm is based on the observation that
+     *           ASCII hex digits have a specific pattern that can be computed efficiently
+     *           using carry flag manipulation.
+     *
+     * @example
+     * <pre>
+     * Input:  0x0123456789ABCDEF
+     * Output: 0x3031323334353637 ('0','1','2','3','4','5','6','7' in ASCII)
```

Comment:
Only the low 32 bits are used, the example only needs to show the input ox 0x0123457.
Or change the signature to accept an `int`.

Reply:
Because the input of Long.expand is long, if the parameter is int, it will cause a redundant conversion process of long -> int -> long.

Reply:
I think the issue is just the example input and output should be something like:
```
Input: 0xABCDEF01
Output: 0x3130666564636261
````
to demonstrate that it is the low-order bits that are used, as well as that the digit order is reversed in the result.